cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -672,6 +698,143 @@ public Route generateExternalBootstrapRoute() {
+                                && broker.getHost() != null)
+                        .map(IngressListenerBrokerConfiguration::getHost)
+                        .findAny()
+                        .orElseThrow(() -> new InvalidResourceException("Hostname for broker with id " + pod + " is required for exposing Kafka cluster using Ingress"));
+
+                dnsAnnotations = listener.getConfiguration().getBrokers().stream()
+                        .filter(broker -> broker != null && broker.getBroker() == pod
+                                && broker.getHost() != null)
+                        .map(IngressListenerBrokerConfiguration::getDnsAnnotations)
+                        .findAny()
+                        .orElse(null);
+            }
+
+            String perPodServiceName = externalServiceName(cluster, pod);
+
+            HTTPIngressPath path = new HTTPIngressPathBuilder()
+                    .withPath("/")
+                    .withNewBackend()
+                        .withNewServicePort(EXTERNAL_PORT)
+                        .withServiceName(perPodServiceName)
+                    .endBackend()
+                    .build();
+
+            IngressRule rule = new IngressRuleBuilder()
+                    .withHost(host)
+                    .withNewHttp()
+                        .withPaths(path)
+                    .endHttp()
+                    .build();
+
+            IngressTLS tls = new IngressTLSBuilder()
+                    .withHosts(host)
+                    .build();
+
+            Ingress ingress = new IngressBuilder()
+                    .withNewMetadata()
+                        .withName(perPodServiceName)
+                        .withLabels(getLabelsWithName(perPodServiceName, templatePerPodIngressLabels))
+                        .withAnnotations(mergeAnnotations(internalAnnotations, templatePerPodIngressAnnotations, dnsAnnotations))
+                        .withNamespace(namespace)
+                        .withOwnerReferences(createOwnerReference())
+                    .endMetadata()
+                    .withNewSpec()
+                        .withRules(rule)
+                        .withTls(tls)
+                    .endSpec()
+                    .build();
+
+            return ingress;
+        }
+
+        return null;
+    }
+
+    /**
+     * Generates a bootstrap ingress which can be used to bootstrap clients outside of Kubernetes.
+     * @return The generated Routes
+     */
+    public Ingress generateExternalBootstrapIngress() {
+        if (isExposedWithIngress()) {
+            Map<String, String> internalAnnotations = new HashMap<>(4);
+            internalAnnotations.put("kubernetes.io/ingress.class", "nginx");
+            internalAnnotations.put("ingress.kubernetes.io/ssl-passthrough", "true");
+            internalAnnotations.put("nginx.ingress.kubernetes.io/ssl-passthrough", "true");
+            internalAnnotations.put("nginx.ingress.kubernetes.io/backend-protocol", "HTTPS");
```

Comment:
factor out a method for these annotations? it's the same code as the `generateExternalIngress` method.